stories/core/FileUpload.stories.tsx

```diff
@@ -0,0 +1,59 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { action } from '@storybook/addon-actions';
+import { useState } from '@storybook/client-api';
+import { Story, Meta } from '@storybook/react';
+import React from 'react';
+import { FileUpload, FileUploadProps, Textarea } from '../../src/core';
+
+export default {
+  component: FileUpload,
+  argTypes: {
+    className: { control: { disable: true } },
+    style: { control: { disable: true } },
+    onFileDropped: { control: { disable: true } },
+    children: { control: { disable: true } },
+  },
+  title: 'Core/FileUpload',
+} as Meta<FileUploadProps>;
+
+export const Basic: Story<FileUploadProps> = (args) => {
+  const [file, setFile] = useState<File | undefined>(undefined);
+
+  return (
+    <FileUpload
+      content={'Drop file to upload'}
+      {...args}
+      onFileDropped={(f) => {
+        setFile(f);
+        action(`${f.name} (${f.size / 1000} KB) uploaded`)();
+      }}
+    >
+      <Textarea placeholder={'Drag a file here'} rows={1} value={file?.name} />
+    </FileUpload>
+  );
+};
+
+Basic.args = {
+  content: 'Drop file to upload',
+};
+
+export const WithoutChildren: Story<FileUploadProps> = ({
+  children,
+  ...rest
+}) => {
+  const [file, setFile] = useState<File | undefined>(undefined);
+
+  return (
+    <FileUpload
+      content={file?.name ?? children ?? 'Drop file to upload'}
```

Comment:
I need to make the `children` include an input element
![image](https://user-images.githubusercontent.com/9084735/116621607-722caa00-a911-11eb-9558-427d59f46630.png)

Not sure how the interaction would work with drop event (which does not need an input element)

Reply:
Actually I think they can stay unrelated, firing separate events 🤔
Will research accessibility approach tomorrow

Reply:
@FlyersPh9 I think we need some focus styling on labels. If not, then we should make them non-focusable. Right now, you can go to `file-upload.html` demo and keep pressing **tab** (to focus without any visual) and **space** (to simulate keypress) to make the file explorer show up for each of those hidden input labels.

Maybe something like this: https://developer.mozilla.org/en-US/docs/Web/API/File/Using_files_from_web_applications#using_a_label_element_to_trigger_a_hidden_file_input_element

Reply:
`:focus-within` has been added in [this PR](https://github.com/iTwin/iTwinUI/pull/54).